src/System.Diagnostics.TraceSource/src/System/Diagnostics/DefaultTraceListener.cs

```diff
@@ -19,31 +20,11 @@ namespace System.Diagnostics
     /// </devdoc>
     public class DefaultTraceListener : TraceListener
     {
-        private class DefaultTraceDebugProvider : DebugProvider
-        {
-            private const int InternalWriteSize = 16384;
+        private static readonly Action<string> s_WriteCore = 
+            (Action<string>)typeof(DebugProvider).
+            GetField("s_WriteCore", BindingFlags.Static | BindingFlags.NonPublic).
+            GetValue(null);
```

Comment:
When I add the new DebugProvider class and Debug.Provider property I can remove s_WriteCore as shown in below diff:

https://github.com/maryamariyan/corefx/pull/31/commits/d1cc83c32a6676e630a117a58576641a3a5c83c0#diff-2d9baaeca242fec8bc2788a0641c0391L23

Since you asked me not to include the new class and APIs to this PR I had to use reflection here.

Reply:
Why can't WriteCore can be a DebugProvider public method that you just call here?

Reply:
Because if user later overrides (current)DebugProvider for WriteCore then they would override OnIndentLevelChanged and other APIs we used for TraceInternal as well (which wire Debug with Trace Listeners).

Reply:
If I were to keep WriteCore in DebugProvider then in DebugTests I couldn't override WriteCore behavior and test TraceInternal behavior at the same time because of the same reason I just mentioned

Reply:
That's why I felt the need for two different classes:
- DebugInternal: only for connecting Debug with TraceListeners
- DebugProvider: only for overriding ShowDialog and Write(Core)

Reply:
> If I were to keep WriteCore in DebugProvider then in DebugTests

To make the old debug tests happy, the first line in WriteCore can be:

`if (s_WriteCore) { s_WriteCore(...); return; }`